pyclaragenomics/install_pyclaragenomics.sh

```diff
@@ -0,0 +1,99 @@
+#!/bin/sh
+
+#
+# Copyright (c) 2019, NVIDIA CORPORATION.  All rights reserved.
+#
+# NVIDIA CORPORATION and its licensors retain all intellectual property
+# and proprietary rights in and to this software, related documentation
+# and any modifications thereto.  Any use, reproduction, disclosure or
+# distribution of this software and related documentation without an express
+# license agreement from NVIDIA CORPORATION is strictly prohibited.
+#
+
+set -e
+
+USAGE_MESSAGE="Usage: $(basename "$0") [-h] [-b cmake_build_folder]
+
+pyclaragenomics installation script.
+
+options:
+    -b , --cmake_build_folder     Set a build folder for Clara Genomics SDK Analysis
+    --help                        Display this help and exit
+    --develop                     Install using pip editble mode
+"
+
+CMAKE_BUILD_FOLDER=""
+PIP_INSTALL_DEVELOP=""
+
+arguments_parser()
+{
+  while [ $# -gt 0 ]; do
+    KEY="$1"
+    case $KEY in
+
+    --help)
+      printf "%s\n" "$USAGE_MESSAGE"
+      exit 0
+      ;;
+
+    -b|--cmake_build_folder)
+      if [ -n "$CMAKE_BUILD_FOLDER" ]; then
+        printf "%s: argument --cmake_build_folder can not be set more than once\n" "$0"
+        exit 1
+      fi
+      shift
+      if [ $# -gt 0 ]; then
+        CMAKE_BUILD_FOLDER="$1"
+      else
+        printf "%s: cmake_build_folder was not specified\n" "$0"
+        exit 1
+      fi
+      shift
+      ;;
+
+    --develop)
+      if [ -n "$PIP_INSTALL_DEVELOP" ]; then
+        printf "%s: argument --develop can no be set more than once\n" "$0"
+        exit 1
+      fi
+      PIP_INSTALL_DEVELOP="-e"
+      shift
+      ;;
+
+    *)
+      printf "Illegal option: %s \n" "${KEY}"
+      printf "%s\n" "$USAGE_MESSAGE"
+      exit 1
+      ;;
+    esac
+  done
+}
+
+# Script Arguments parsing
+arguments_parser "$@"
+
+# Validate required argument
+if [ -z "$CMAKE_BUILD_FOLDER" ]; then
+  printf ' %s: -b , --cmake_build_folder argument is mandatory\n' "$0"
+  exit 1
+fi
+
+# Create build folder if does not exist
+if [ ! -e "$CMAKE_BUILD_FOLDER" ]; then
+  mkdir -p "$CMAKE_BUILD_FOLDER"
+elif [ ! -d "$CMAKE_BUILD_FOLDER" ]; then
+    printf "%s already exists, but is not a directory\n" "$CMAKE_BUILD_FOLDER"
+fi
+
+# Build & install Clara Genomics Analysis SDK
+python build_clara_genomics_analysis.py --build_output_folder "${CMAKE_BUILD_FOLDER}/cga_build"
```

Comment:
I'm wondering if we need this python script to wrap cmake anymore. Originally we had the cmake building wrapper in python to keep it colocate with the `setup` call. But now that it's separate, we could just call cmake directly in bash.

We would need to add some special handling for checking if the CUDA_TOOLKIT_DIR env var is set, but other than that it should be a single liner really.

Reply:
Or we get rid of the shell script and wrap it all in another python script which internally calls `pip install ...`

Reply:
Got rid of the wrapper shell script.